[RHIDP-11676]: Add log aggregation and OpenTelemetry documentation (#2082)
redhat-developer/red-hat-developers-documentation-rhdh: modules/extend_orchestrator-in-rhdh/ref-trace-context-propagation-headers-across-services.adoc

:_mod-docs-content-type: REFERENCE

[id="trace-context-propagation-headers-across-services_{context}"]
= Trace context propagation headers across services

[role="_abstract"]
HTTP headers that SonataFlow extracts and propagates to maintain trace correlation across service boundaries. Use these headers to link workflow executions in distributed transactions and pass custom tracking context through service chains.

SonataFlow extracts and propagates the following headers to maintain observability across service boundaries:

> * `X-TRANSACTION-ID`: Correlates multiple workflow executions that belong to the same broader business transaction.

Reviewer: ```suggestion
* `X-TRANSACTION-ID`: Correlates multiple workflow executions that belong to the same business transaction.
```

> * `X-TRACKER-{asterisk}`: Any custom tracking context provided in these headers is sanitized and automatically converted into span attributes, for example, `sonataflow.tracker.{asterisk}`, for easier querying.

Reviewer: ```suggestion
* `X-TRACKER-{asterisk}`: Sanitizes and converts any custom tracking context from headers into span attributes, such as `sonataflow.tracker.{asterisk}`, to simplify querying.
```
